autocomplete/bash_autocomplete

```diff
@@ -1,23 +1,23 @@
-#! /bin/bash
+#!/usr/env/bin bash
```

Comment:
Why? Maybe it is obvious, but I am not sure. In Python we use `#!/usr/env/bin python`, so that scripts can be run from `vitrualenv` where `python` points to different binary with different set of libs. What is the point here with `bash`?

Reply:
I was brought up in the practice of using `/usr/bin/env bash,` and I believe it's a good shebang, because it doesn't hardcode the path. [See this answer](https://stackoverflow.com/a/16365367/7009800).

Reply:
There is a mistake in new shebang, and that makes me think that you haven't tested that it actually works. :D

Using `env` has drawbacks. I still don't see why it is so good. Where do you expect hardcoded `/bin/bash` to fail?

Reply:
mistake fixed, thanks.

Reply:
also, what kind of drawbacks?

Reply:
> also, what kind of drawbacks?

`bash` then can point to any binary. Any user or malicious script can override it. `/bin/bash` is guaranteed to be present if `bash` is installed. So it is just less moving parts, less attack surface, and less things to care about.

If you can ask your mentor about the reasons why he/she prefers `env` calls, maybe he can actually remember the case where `env` with `bash` was really necessary.

Reply:
I see. I reverted back to `/bin/bash` (minus the additional whitespace).